src/NuGet.Core/NuGet.Protocol/Utility/MetadataReferenceCache.cs

```diff
@@ -97,12 +100,28 @@ public T GetObject<T>(T input)
                 _propertyCache.Add(typeKey, properties);
             }
 
+            if (!_stringMethodsCache.ContainsKey(typeof(MetadataReferenceCache)))
+            {
+                // Doing reflection everytime is expensive so cache it for string type which is all this MetadataReferenceCache about.
+                var stringPropertyType = typeof(string);
+                var method = _metadataReferenceCacheType.GetTypeInfo()
+                        .DeclaredMethods.FirstOrDefault(
+                            m =>
+                                m.Name == CachableTypesMap[stringPropertyType] &&
+                                m.GetParameters().Select(p => p.ParameterType).SequenceEqual(new Type[] { stringPropertyType }));
+                _stringMethodsCache.Add(_metadataReferenceCacheType, method);
+            }
+
             for (var i=0; i < properties.Length; i++)
             {
                 var property = properties[i];
 
                 var value = property.GetMethod.Invoke(input, null);
-                var cachedValue =
+
+                var cachedValue = property.PropertyType == typeof(string) ?
```

Comment:
These probably pre-dated our coding guideline. Please change `var` for these.

Reply:
Fixed